mypy/test/teststubgen.py

```diff
@@ -904,11 +961,54 @@ def test(cls, arg0: str) -> None:
         imports: list[str] = []
         mod = ModuleType(TestClass.__module__, "")
         generate_c_function_stub(
-            mod, "test", TestClass.test, output, imports, self_var="cls", class_name="TestClass"
+            mod,
+            "test",
+            TestClass.test,
+            output,
+            imports,
+            self_var="cls",
+            class_name="TestClass",
+            sig_generators=get_sig_generators(parse_options([])),
         )
-        assert_equal(output, ["def test(cls, *args, **kwargs) -> Any: ..."])
+        assert_equal(output, ["@classmethod", "def test(cls, *args, **kwargs) -> Any: ..."])
         assert_equal(imports, [])
 
+    def test_generate_c_type_classmethod_with_overloads(self) -> None:
+        class TestClass:
+            @classmethod
+            def test(self, arg0: str) -> None:
+                """
+                test(cls, arg0: str)
+                test(cls, arg0: int)
+                """
+                pass
+
+        output: list[str] = []
+        imports: list[str] = []
+        mod = ModuleType(TestClass.__module__, "")
+        generate_c_function_stub(
+            mod,
+            "test",
+            TestClass.test,
+            output,
+            imports,
+            self_var="cls",
+            class_name="TestClass",
+            sig_generators=get_sig_generators(parse_options([])),
+        )
+        assert_equal(
+            output,
+            [
+                "@classmethod",
```

Comment:
Hmm, I think this classmethod + overload ordering is unconventional? Should be trivial to switch

Reply:
I'm fine with this suggestion, but note that this will change the order that these were being applied. 

```python

Reply:
Sorry, I hit enter too soon.   Do you think it's worth changing this if it will change the existing behavior?